skimage/filters/rank/generic.py

```diff
@@ -63,17 +62,59 @@
            'entropy', 'otsu']
 
 
-def _handle_input(image, selem, out, mask, out_dtype=None, pixel_size=1):
+def _handle_input(image, selem, out=None, mask=None, out_dtype=None,
+                  pixel_size=1):
+    """Preprocess and verify input for filters.rank methods.
+
+    Parameters
+    ----------
+    image : 2-D array (integer, float or boolean)
+        Input image.
+    selem : 2-D array (integer, float or boolean)
+        The neighborhood expressed as a 2-D array of 1's and 0's.
+    out : 2-D array (integer, float, boolean or optional)
+        If None, a new array is allocated.
+    mask : ndarray (integer, float, boolean or optional)
+        Mask array that defines (>0) area of the image included in the local
+        neighborhood. If None, the complete image is used (default).
+    out_dtype : data-type, optional
+        Desired output data-type. Default is None, which means we cast output
+        in input dtype.
+    pixel_size : int, optional
+        Dimension of each pixel. Default value is 1.
+
+    Returns
+    -------
+    image : 2-D array (np.uint8 or np.uint16)
+    selem : 2-D array (np.uint8)
+        The neighborhood expressed as a binary 2-D array.
+    out : 3-D array (same dtype out_dtype or as input)
+        Output array. The two first dimensions are the spatial ones, the third
+        one is the pixel vector (length 1 by default).
+    mask : 2-D array (np.uint8)
+        Mask array that defines (>0) area of the image included in the local
+        neighborhood.
+    n_bins : int
+        Number of histogram bins.
+    out_dtype : data-type
```

Comment:
Did you mean to return this? It doesn't seem to be used anywhere.

Reply:
Looking at the commit history, it seems that there was a point in time where it was useful to return this, now it no-longer seems to be the case. I was able to remove the output, refactor the code, and it passes all rank tests now